<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="content-type" content="text/html; charset=utf-8" />
		<meta name="description" content="Your description goes here" />
		<meta name="keywords" content="Ferdinando Fioretto,Nando Fioretto,Nando,Ferdinando,Fioretto,DCOP,Smart Grid,AI,GPU,Distributed" />
		<meta name="author" content="Ferdinando Fioretto" />
		<link rel="shortcut icon" href="img/su_icon.png">
	<link href="https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz" rel="stylesheet">
		<link rel="stylesheet" type="text/css" media="all" href="inland.css" />
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
		<script type="text/javascript" src="js/jquery.nivo.slider.js"></script>
		<title>Ferdinando Fioretto</title>
		<script>
			$(document).ready(function(){
				var moreText = "Show more",
				lessText = "Show less",
				moreButton = $("a.read_more");

				moreButton.click(
					function () {
						var $this = $(this);
						$this.text($this.text() == moreText ? lessText : moreText).next("div.more").slideToggle("fast");
					}
					);
			});
		</script>
		<script>
			$(document).ready(function(){
				var moreText = "Show more",
				lessText = "Show less",
				moreButton = $("a.read_more_title");
				moreButton.click(
					function () {
						$(this).nextAll("div.more").slideToggle("fast");
						var $show_text = $(this).nextAll("a.read_more");
						$show_text.text($show_text.text() == moreText ? lessText : moreText)
					}
					);
			});
		</script>
	</head>

	<body>
		<div id="wrapper960" class="clearfix">
			<div id="toplinks">
				<ul class="toplinks_links">
					<li>
						<a href='https://www.isye.gatech.edu'>
						<img align='center' src="img/su_logo.png"  height="40pt" style="margin:-14px -20px -14px 0px">
						</a>
					</li>
				</ul>
			</div>

			<div id="header" class="clearfix shadow">
				<div id="sitetitle" class="clearfix">
					<h1>Ferdinando Fioretto</h1>
				</div>

			<div id="nav" class="clearfix">
				<ul>
					<li><a href="index.html">Home</a></li>
					<!-- <li><a href="news.html">News</a></li> -->
					<li><a class="current" href="research.html">Research</a></li>
					<li><a href="publications.html">Publications</a></li>
					<li><a href="people.html">People</a></li>
					<li><a href="teaching.html">Teaching</a></li>
					<li><a href="index.html#contacts">Contacts</a></li>
				</ul>
			</div>
		</div>

		<!-- HEAD -->
		<div id="content" class="clearfix shadow">
			<h1>Computational Biology</h1>
		</div>

		<div id="content">
		<table style="height: 220px;" width="100%">
			<tr>
				<td style="vertical-align: top;" align="justified">
					<p style="text-align: justify; padding-bottom: 0px">
					My research on computational biology has focused on two main topics:

					<ul>
						<li> <strong>Gene Regulatory Network </strong> <br>
						A Gene Regulatory Networks (GRN) is a collection of DNA segments that interact with each other either directly or indirectly (e.g., through their RNA and protein expression products). Understanding the behavior of GRNs is of core interest in computational biology, as it is closely related with the process of drug production and assessment.
						My research has focused on augmenting machine learning procedures with constraint optimization techniques to predict GRN structures. I have investigated how to use committee machines to predict possible GRNs solutions, and augmented them with biologically meaningful constraints to filter irrelevant genes interactions.
					</li>

					<li> <strong>Protein Structure Prediction</strong> <br>
						Proteins are the product of coding genes, and perform a vast number of functions within living organisms. Their activity is strongly related with their spatial conformation, therefore the Protein Structure Prediction problem, that aims at predicting the 3D structure of a protein given its chemical composition, is of central importance.
						My work has focused on the development of a specialized constraint solver for finding meaningful protein 3D conformations.
						My work has been used for a variety of different structural protein analysis, including loop modeling and structure prediction. This technology can be used to support a range of applications, including  protein-to-protein binding, protein flexibility study, and structure prediction.
					</li>
				</ul>
 				</p>
				</td>
				<td style="vertical-align: top; padding-left: 10px"  width="200">
					<br>
					<img src="img/dna.jpg" alt="Computational Biology" width="200" height="150"/>
				</td>

			</tr>
		</table>

		<h3>Related Awards</h3>
		<ul>
		    <li>
		      <span style="color: #000;">Best Student Paper Award</span>: Computational Methods in Systems Biology (CMSB), 2013. <br>
		      Ferdinando Fioretto and Enrico Pontelli.
	    	  <a href="publications.html">&quot;Constraint Programming in Community-based Gene Regulatory Network Inference&quot;</a>.
	    	</li>
		</ul>

		<h3>Collaborators</h3>
		<ul>
			<li><a href="https://www.linkedin.com/in/federicocampeotto/?locale=en_US">Federico Campeotto</a>, Amazon </li>
			<li><a href="Agostino Dovier">Agostino Dovier</a>, University of Udine</li>
			<li><a href="http://www2.unipr.it/~dalpalu/">Alessandro Dal Palu'</a>, University of Parma</li>
			<li><a href="https://www.cs.nmsu.edu/wp/people/faculty-members/enrico-pontelli/">Enrico Pontelli</a>, New Mexico State University </li>
		</ul>

		<h3>Relevant Publications</h3>

		<ul>
			<!-- tomacs15 -->
			<li name="tomacs15" id="tomacs15">
				<strong>Ferdinando Fioretto</strong>, Agostino Dovier and Enrico Pontelli.
				<a class="read_more_title">&quot;Constrained Community-based Gene Regulatory Network Inference&quot;</a>.
				In <em>ACM Transactions on Modeling and Computer Simulation (TOMACS)</em>, 2015.
				<br>
				<span class="paper_links">
					Downloads:
					<a href="files/papers/tomacs15.pdf"  target="blank">[pdf]</a>
					<a href="files/papers/tomacs15_appendix.pdf" target="blank">[appendix]</a>
					<a href="files/bibtex/tomacs15-bib.txt" target="blank">[BibTex]</a>
					| Links:
					<a href="http://dl.acm.org/citation.cfm?doid=2688909" target="blank">[web]</a>
					<a href="https://github.com/nandofioretto/CCN-GRN"  target="blank">[github]</a>
			  	</span>
			  	<a class="read_more">Show more</a>
			  	<div class="more abstract bg-blue">
			  		<strong><em>Abstract</em></strong>:
			  		The problem of gene regulatory network inference is a major concern of systems biology. In recent years, a novel methodology has gained momentum, called community network approach. Community networks integrate predictions from individual methods in a “metapredictor,” in order to compose the advantages of different methods and soften individual limitations. This article proposes a novel methodology to integrate prediction ensembles using constraint programming, a declarative modeling and problem solving paradigm. Constraint programming naturally allows the modeling of dependencies among components of the problem as constraints, facilitating the integration and use of different forms of knowledge. The new paradigm, referred to as constrained community network, uses constraints to capture properties of the regulatory networks (e.g., topological properties) and to guide the integration of knowledge derived from different families of network predictions. The article experimentally shows the potential of this approach: The addition of biological constraints can offer significant improvements in prediction accuracy.
			  	</div>
		    </li>


			<!-- WCB -->
			<li name="wcb14" id="wcb14">
				Federico Campeotto, Alessandro Dal Palu', Agostino Dovier, <strong>Ferdinando Fioretto</strong>, Enrico Pontelli.
				<a class="read_more_title">&quot;Experimenting with FIASCO for protein structure prediction&quot;</a>.
				In <em>Workshop on Constraint-Based Methods for Bioinformatics (WCB)</em>, 2014
				<br>
				<span class="paper_links">
					Downloads:
					<a href="files/papers/wcb14.pdf"  target="blank">[pdf]</a>
<!-- 									<a href="files/talks/wcb14-slides.pdf" target="blank">[slides]</a>
--> 									<a href="files/bibtex/wcb14-bib.txt" target="blank">[BibTex]</a>
					| Links:
					<a href="http://cp2014.a4cp.org/workshops/bioinfo14" target="blank">[web]</a>
					<a href="mailto:fioretto@umich.edu?subject=Code request for paper id [wcb14]
					&body=Hi,  I would like to request code and results associated to your paper ID: wcb14. Thanks!
					&cc=epontell@cs.nmsu.edu">[request code]</a>
			  	</span>
			  	<a class="read_more">Show more</a>
			  	<div class="more abstract bg-blue">
			  		<strong><em>Abstract</em></strong>:
					Constraint Programming (CP) [6] is a declarative programming methodology that has gained a predominant role in addressing large scale combinatorial and
					optimization problems. The CP paradigm provides the tools necessary to guide the modeling and resolution of search problems. In particular, it enables the declarative modeling (in terms of variables and constraints) of problems, the ability to rapidly propagate the effects of search decisions, and flexible and efficient procedures to explore the search space of possible solutions. The focus of this work is the use of constraint-based technology in the investigation of protein structures. We developed a system composed of two main parts: (1) a Graphical User Interface, to help the user, in particular users coming from Biology, in modeling protein structure studies with CP, and (2) a constraint solver targeted at protein structure analysis. Parts of the system have been presented in four previous works. In [1] a first prototype of the graphical interface has been presented, using a preliminary version of the solver later presented in [2] and, still improved, in [5]. This solver focused on one class of constraints targeting the problem of loop closure; our current work provides instead a comprehensive constraint system for modeling generic structural protein properties and investigating different types of problems (e.g., structure prediction, studies of flexibility). Moreover the solver makes use, in some parts, of GPU computation following the ideas presented in [4], as explained below.
				</div>
			</li>

			<li name="jair13" id="jair13">
				Federico Campeotto, Alessandro Dal Palu', Agostino Dovier, <strong>Ferdinando Fioretto</strong>, Enrico Pontelli.
				<a class="read_more_title">&quot;A Constraint Solver for Flexible Protein Models&quot;</a>.
				In <em>Journal of Artificial Intelligence Research (JAIR)</em>, 2013.
				<br>
				<span class="paper_links">
					Downloads:
					<a href="files/papers/jair13.pdf"  target="blank">[pdf]</a>
					<a href="files/bibtex/jair13-bib.txt" target="blank">[BibTex]</a>
					| Links:
					<a href="https://jair.org/index.php/jair/article/view/10856" target="blank">[web]</a>
				</span>
				<a class="read_more">Show more</a>
				<div class="more abstract bg-blue">
					<strong><em>Abstract</em></strong>:
					This paper proposes the formalization and implementation of a novel class of constraints aimed at modeling problems related to placement of multi-body systems in the 3-dimensional space. Each multi-body is a system composed of body elements, connected by joint relationships and constrained by geometric properties. The emphasis of this investigation is the use of multi-body systems to model native conformations of protein structures---where each body represents an entity of the protein (e.g., an amino acid, a small peptide) and the geometric constraints are related to the spatial properties of the composing atoms. The paper explores the use of the proposed class of constraints to support a variety of different structural analysis of proteins, such as loop modeling and structure prediction.
					The declarative nature of a constraint-based encoding provides elaboration tolerance and the ability to make use of any additional knowledge in the analysis studies. The filtering capabilities of the proposed constraints also allow to control the number of representative solutions that are withdrawn from the conformational space of the protein, by means of criteria driven by uniform distribution sampling principles. In this scenario it is possible to select the desired degree of precision and/or number of solutions. The filtering component automatically excludes configurations that violate the spatial and geometric properties of the composing multi-body system. The paper illustrates the implementation of a constraint solver based on the multi-body perspective and its empirical evaluation on protein structure analysis problems.
				</div>
			</li>

			<!-- CMSB 2013 -->
			<li name="cmsb13" id="cmsb13">
				<strong>Ferdinando Fioretto</strong>, Enrico Pontelli.
				<a class="read_more_title">&quot;Constraint Programming in Community-based Gene Regulatory Network Inference&quot;</a>.
				In <em>Proceedings of the Computational Methods in System Biology (CMSB)</em>, 2013.
				<br>
				<span class="paper_links">
					Downloads:
					<a href="files/papers/cmsb13.pdf"  target="blank">[pdf]</a>
						<a href="files/talks/cmsb13-slides.pdf"  target="blank">[slides]</a>
						<a href="files/bibtex/cmsb13-bib.txt" target="blank">[BibTex]</a>
					| Links:
					<a href="http://ist.ac.at/cmsb13/program/" target="blank">[web]</a>
					<a href="mailto:fioretto@umich.edu?subject=Code request for paper id [cmsb13]
					&body=Hi,  I would like to request code and results associated to your paper ID: cmsb13. Thanks!
					&cc=epontell@cs.nmsu.edu">[request code]</a>
				</span>
				<span style="font-size: 105%; margin: 0 0 0 4pt; color: #B03A2E;">*Best student paper award*</span>
				<a class="read_more">Show more</a>
				<div class="more abstract bg-blue">
					<strong><em>Abstract</em></strong>:
					Gene Regulatory Network (GRN) inference is a major objective of Systems Biology. The complexity of biological systems and the lack of adequate data have posed many challenges to the inference problem. Community networks integrate predictions from individual methods in a “meta predictor”, in order to compose the advantages of different methods and soften individual limitations. This paper proposes a novel methodology to integrate prediction ensembles using Constraint Programming, a declarative modeling paradigm, which allows the formulation of dependencies among components of the problem, enabling the integration of diverse forms of knowledge. The paper experimentally shows the potential of this method: the addition of biological constraints can offer improvements in the prediction accuracy, and the method shows promising results in assessing biological hypothesis using constraints.
				</div>
			</li>

			<li name="cp12" id="cp12">
				Federico Campeotto, Alessandro Dal Palu', Agostino Dovier, <strong>Ferdinando Fioretto</strong> and Enrico Pontelli.
				<a class="read_more_title">&quot;A Filtering Technique for Fragment Assembly-based Proteins Loop Modeling with Constraints&quot;</a>.
				In <em>Proceedings of the International Conference on Principles and Practice of Constraint Programming (CP)</em>, 2012.
				<br>
				<span class="paper_links">
					Downloads:
					<a href="files/papers/cp12.pdf"  target="blank">[pdf]</a>
	<!-- 									<a href="files/talks/cp12-slides.pdf"  target="blank">[slides]</a>
	-->									<a href="files/bibtex/cp12-bib.txt" target="blank">[BibTex]</a>
					| Links:
					<a href="https://link.springer.com/chapter/10.1007/978-3-642-33558-7_61" target="blank">[web]</a>
					<a href="mailto:fioretto@umich.edu?subject=Code request for paper id [cp12]
					&body=Hi,  I would like to request code and results associated to your paper ID: cp12. Thanks!
					&cc=epontell@cs.nmsu.edu">[request code]</a>
				</span>
				<a class="read_more">Show more</a>
				<div class="more abstract bg-blue">
					<strong><em>Abstract</em></strong>:
					Methods to predict the structure of a protein often rely on the knowledge of macro sub-structures and their exact or approximated relative positions in space. The parts connecting these sub-structures are called loops and, in general, they are characterized by a high degree of freedom. The modeling of loops is a critical problem in predicting protein conformations that are biologically realistic. This paper introduces a class of constraints that models a general multi-body system; we present a proof of NP-completeness and provide filtering techniques, inspired by inverse kinematics, that can drastically reduce the search space of potential conformations. The paper shows the application of the constraint in solving the protein loop modeling problem, based on fragments assembly.
				</div>
			</li>

		</ul>

		</div>

		<!-- Footer -->
		<div id="footer" class="shadow">
		  <p>&copy; 2018 - 2020 Ferdinando Fioretto | <a href="http://andreasviklund.com/templates/inland/">Template design</a> by <a href="http://andreasviklund.com/">andreasviklund.com</a><br /></p>
		  <p> Last Update: Mar. 2020
		</div>
	</div>

	<script type="text/javascript">
				$(window).load(function() {
					$('#slider').nivoSlider();
				});
	</script>

	<script>w3.includeHTML();</script>

	</body>
</html>
